cluster/src/main/java/com/hartwig/pipeline/tertiary/lilac/LilacBamSlicer.java

```diff
@@ -0,0 +1,146 @@
+                slicedBam(metadata.reference().sampleName())).stream(),
+                buildCommands(getTumorBamDownload().getLocalTargetPath(), slicedBam(metadata.tumor().sampleName())).stream())
+                .collect(toList());
+
+    }
+
+    @Override
+    public List<BashCommand> referenceOnlyCommands(final SomaticRunMetadata metadata) {
+        return buildCommands(getReferenceBamDownload().getLocalTargetPath(), metadata.sampleName());
+
+    }
+
+    @Override
+    public List<BashCommand> tumorOnlyCommands(final SomaticRunMetadata metadata) {
+        return buildCommands(getTumorBamDownload().getLocalTargetPath(), metadata.sampleName());
+    }
+
+    @Override
+    public VirtualMachineJobDefinition vmDefinition(final BashStartupScript bash, final ResultsDirectory resultsDirectory) {
+        return ImmutableVirtualMachineJobDefinition.builder()
+                .name(NAMESPACE)
+                .startupCommand(bash)
+                .performanceProfile(VirtualMachinePerformanceProfile.custom(8, 16))
+                .namespacedResults(resultsDirectory)
+                .build();
+    }
+
+    @Override
+    public LilacBamSliceOutput output(final SomaticRunMetadata metadata, final PipelineStatus jobStatus, final RuntimeBucket bucket,
+            final ResultsDirectory resultsDirectory) {
+        return LilacBamSliceOutput.builder()
+                .status(jobStatus)
+                .addFailedLogLocations(GoogleStorageLocation.of(bucket.name(), RunLogComponent.LOG_FILE))
+                .addReportComponents(new EntireOutputComponent(bucket, Folder.root(), namespace(), resultsDirectory))
+                .addDatatypes(datatypes(metadata))
+                .build();
+    }
+
+    @Override
+    public LilacBamSliceOutput skippedOutput(final SomaticRunMetadata metadata) {
+        return LilacBamSliceOutput.builder().status(PipelineStatus.SKIPPED).build();
+    }
+
+    @Override
+    public LilacBamSliceOutput persistedOutput(final SomaticRunMetadata metadata) {
+        return LilacBamSliceOutput.builder().status(PipelineStatus.PERSISTED).addDatatypes(datatypes(metadata)).build();
+    }
+
+    @Override
+    public boolean shouldRun(final Arguments arguments) {
+        return arguments.runTertiary() && !arguments.shallow();
+    }
+
+    private List<BashCommand> buildCommands(final String inputBamFile, final String slicedBamFile) {
+        return List.of(new SambambaCommand("slice", "-L", resourceFiles.hlaRegionBed(), "-o", slicedBamFile, inputBamFile),
+                new SambambaCommand("index", slicedBamFile));
+    }
+
+    private String slicedBam(final String sampleName) {
+        return VmDirectories.outputFile(sampleName + ".hla.bam");
+    }
+
+    private String bai(final String bam) {
+        return bam + ".bai";
+    }
+
+    private AddDatatype[] datatypes(final SomaticRunMetadata metadata) {
+        ArrayList<AddDatatype> datatypes = new ArrayList<>();
+        if (metadata.isSingleSample()) {
```

Comment:
Maybe better to use the maybeReferenceSample and maybeTumorSample to decide whether to include the datatype.